File: engine/model/src/main/scala/tools/packet/ui/GetMacros.scala
package tools.packet.ui

import net.opcodes.SendOpcode
import net.server.SkillMacro
import tools.packet.PacketInput

class GetMacros(private var _macros: Array[SkillMacro]) extends PacketInput {
  def macros: Array[SkillMacro] = _macros

  override def opcode(): SendOpcode = SendOpcode.MACRO_SYS_DATA_INIT
}

File: engine/engine-model/src/main/java/tools/packet/ui/GetMacros.java
package tools.packet.ui;

import net.opcodes.SendOpcode;
import net.server.SkillMacro;
import tools.packet.PacketInput;

public record GetMacros(SkillMacro[] macros) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MACRO_SYS_DATA_INIT;
   }
}